### PR TITLE
Renovation: Move viewModel to class with component declaration

### PR DESCRIPTION
**Pull request by roman-simionov**

### CODE REVIEW

**Review by pavelgruba — DISMISSED** [review]

File: js/renovation/button.tsx [hunk 1]
```diff
@@ -46,80 +46,52 @@ const getAriaLabel = (text, icon) => {
...
-    const renderText = !viewModel.contentRender && viewModel.text;
-    const isIconLeft = viewModel.iconPosition === 'left';
-    const leftIcon = !viewModel.contentRender && isIconLeft;
-    const rightIcon = !viewModel.contentRender && !isIconLeft;
-    const icon = !viewModel.contentRender && viewModel.iconSource
-        && <Icon source={viewModel.iconSource} position={viewModel.iconPosition}/>;
+export const viewFunction = (viewModel: Button) => {
+    const renderText = !viewModel.props.contentRender && viewModel.props.text;
+    const isIconLeft = viewModel.props.iconPosition === 'left';
+    const leftIcon = !viewModel.props.contentRender && isIconLeft;
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: Why don't you use object destruction?

```js
({ props: { text, ... }, ... })
```

or

```js
const { text, ... } = viewModel.props;
```

**Review by pavelgruba — DISMISSED** [review]

File: js/renovation/button.tsx [hunk 2]
```diff
@@ -207,4 +178,20 @@ export default class Button extends JSXComponent<ButtonInput> {
...
+    get cssClasses():string {
+        return getCssClasses(this.props);
+    }
+
+    get elementAttr() {
+        return { ...this.props.elementAttr, role: 'button' };
+    }
+
+    get iconSource():string {
+        return (this.props.icon || this.props.type === 'back') ? (this.props.icon || 'back') : '';
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: const { icon, type } = this.props;

File: js/renovation/button.tsx (hunk 1 above)
Comment (roman-simionov) [reply js/renovation/button.tsx]: I did not write this code. Refactoring goes out of the context of this PR

**Review by pavelgruba — DISMISSED** [review]

File: js/renovation/button.tsx (hunk 1 above)
Comment (MaximKudriavtsev) [reply js/renovation/button.tsx]: Okay

**Review by MaximKudriavtsev — APPROVED** [review]

**Pull request merged**